docs/tutorial-3.md

```diff
@@ -0,0 +1,162 @@
+# How to include external Actions
+
+Now an [example rust project](https://github.com/input-output-hk/cicero-example/tree/main/rust) is used, to explain how to setup an external repository with Cicero.
+
+We will call this an external Action, because the Action will be provided by a repository which isn't the Cicero repository itself.
+
+First the repository needs a [flake.nix](https://github.com/input-output-hk/cicero-example/blob/main/rust/flake.nix) containing **Cicero** itself as flake input and **ciceroActions** as flake output.
+
+## flake.nix:
+```nix
+{
+  description = "Flake for Cicero rustExample";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    utils.url = "github:numtide/flake-utils";
+    cicero = {
+      url = "github:input-output-hk/cicero";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, utils, cicero }:
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in
+      {
+        defaultPackage = pkgs.stdenv.mkDerivation {
+          name = "rustExample";
+          src = ./src;
+          buildInputs = with pkgs; [
+            coreutils
+            gcc
+            rustc
+          ];
+          installPhase = ''
+            rustc $src/hello.rs
+            mkdir -p $out
+            mv ./hello $out
+          '';
+        };
+      }) // {
+        ciceroActions = cicero.lib.callActionsWithExtraArgs
+          rec {
+            inherit (cicero.lib) std;
+            inherit (nixpkgs) lib;
+            actionLib = import cicero/lib.nix { inherit cicero lib; };
+            nixpkgsRev = nixpkgs.rev;
+          } ./cicero/actions;
+      };
+}
+```
+
+## Writing the external Action
+
+Now it's required to write the corresponding Action for Cicero to execute later on.
+
+The ciceroActions already hints that it does a recursive lookup in the ./cicero folder, so therefore it's possible to place our Action under the ./cicero/actions/rust/example directory.
+
+### ci.nix
+The Action itself is written as nix expression.
+
+```
+// Those are the inputs to the actual Action function
+// The path of the file from the starting directory is used as name.
+// std, lib & actionLib are provided by the set/rec in the ciceroActions flake output.
+{ name, std, lib, actionLib, ... } @ args:
+
+{
+
+  // The inputs.start describes the expected Fact as json input
+  // expecting sha & clone_url as fields
+  // while statuses_url, ref & default_branch being optional fields
+  inputs.start = ''
+    "rust/example": start: {
```

Comment:
Is `chezscheme/example` action name? If so can we add this information to book

Reply:
rust/example/build Is the actual action name.

rust/example is just a name set by the input fact description.